src/app/app-designs/app-designs.component.ts

```diff
@@ -70,6 +83,50 @@ export class AppDesignsComponent implements OnInit, OnDestroy {
     this.mapObservable = this.store.pipe(selectNotNull((store) => store.interactiveMap.selectedMap)).subscribe((map) => {
       this.store.dispatch(new SetMap(map));
     });
+
+    this.dataSource.sortingDataAccessor = (item, property) => {
+      switch (property) {
+        case 'reaction_knockins':
+        case 'reaction_knockouts':
+        case 'gene_knockouts':
+          return item['design'][property].length;
+        case 'organism_id':
+          return item['model'][property];
+        default:
+          return item[property];
+      }
+    };
+
+    this.collapseClicked.subscribe((value) => {
+      this.expandedId = value ? value.id : null;
+    });
+  }
+
+  getModelName(modelId: number): Observable<string> {
+    return this.store.pipe(select(getModelName(modelId)));
+  }
+
+  getOrganismName(organismId: number): Observable<string> {
+    return this.store.pipe(select(getOrganismName(organismId)));
+  }
+
+  geneLink(geneId: string): string {
+    return `http://bigg.ucsd.edu/search?query=${geneId}`;
+  }
+
+  reactionLink(reaction: string, method: string, isAddedReactions: boolean): string {
+    const reactionId = (isAddedReactions && method === 'Pathway')
+      ? JSON.parse(reaction).bigg_id
+      : reaction;
+    if (reactionId.startsWith('MNX')) {
+      return `https://www.metanetx.org/equa_info/${reactionId}`;
+    } return `http://bigg.ucsd.edu/search?query=${reactionId}`;
+  }
+
+  getReactionId(reaction: string, method: string): string {
+    if (method === 'Pathway') {
+      return JSON.parse(reaction).bigg_id;
+    } return reaction;
```

Comment:
I think we are duplicating all these functions, so maybe is better if we include these functions on `app/lib.ts` and we can import it from there in both components